tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/MiscellaneousRuleTests.cs

```diff
@@ -214,6 +215,25 @@ public void PropertiesDataSourcesMustMatchItemDataSources(string ruleName, strin
             }
         }
 
+        [Theory]
+        [MemberData(nameof(GetAllRules))]
+        public void PropertyNamesMustNotContainSpaces(string ruleName, string fullPath)
+        {
+            // While MSBuild properties may not contain spaces in their names this restriction doesn't necessarily
+            // apply when the property in the Rule connects to a non-MSBuild DataSource. Currently none of our
+            // properties have names with spaces, so for the moment we'll just keep this test simple.
+            
+            var root = LoadXamlRule(fullPath);
+
+            foreach (var property in GetProperties(root))
+            {
+                var name = property.Attribute("Name")?.Value;
+
+                Assert.NotNull(name);
+                Assert.False(name.Contains(" "), $"Property name '{name}' in rule '{ruleName}' contains a space. This is likely to lead to a runtime exception when accessing the property");
```

Comment:
We could be more assertive here.

From https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-properties?view=vs-2022#define-and-reference-properties-in-a-project-file:

> Valid property names begin with an uppercase or lowercase letter or underscore (_); valid subsequent characters include alphanumeric characters (letters or digits), underscore, and hyphen (-).

Reply:
I was thinking this too, but I didn't look for that reference doc on what the actual valid characters are. A nice regex would be an easy way to test it.